.config/nvim/lua/plugins/codecompanion.lua

```diff
@@ -29,6 +29,19 @@ return {
       local layout = vim.env.CC_LAYOUT_OVERRIDE or "vertical"
       return {
         adapters = {
+          acp = {
+            opencode = function()
+              return require("codecompanion.adapters").extend("acp", {
+                name = "opencode",
+                url = "http://localhost:3210/acp",
```

Comment:
Have you even checked the existing implementations?

https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/adapters/acp/gemini_cli.lua

There is no URL. Also would make much sense. You have to invoke the introduces cli command 'opencode acp'

Try again...